### PR TITLE
Drop support for .NET 6/7 (Lombiq Technologies: OCORE-128)

### LINKED ISSUE #14694: Drop support for .NET 6/7
Is there a reason to keep support for .NET 6 and .NET 7 that .NET 8 is already supported?

@jtkech @agriffard  @Piedone @Skrypt any objections?

### PR DESCRIPTION
**Pull request by Piedone**
Fixes #14694

- Functional tests also pass: https://github.com/OrchardCMS/OrchardCore/actions/runs/6870220593
- The "Found version-specific or distribution-specific runtime identifier(s): alpine-arm, alpine-arm64, alpine-x64. Affected libraries: SQLitePCLRaw.lib.e_sqlite3. In .NET 8.0 and higher, assets for version-specific and distribution-specific runtime identifiers will not be found by default. See https://aka.ms/dotnet/rid-usage for details." build warnings need to be fixed in YesSql: https://github.com/sebastienros/yessql/issues/510.

### CODE REVIEW

File: .github/workflows/functional_all_db.yml
Comment (Piedone) [inline .github/workflows/functional_all_db.yml:3]: This needs to be reverted.

**Comment by hishamco** [comment]
I'm not sure why we need to drop the support for both .NET 6 & 7? OC supports multi-targeting from the beginning

We need to drop .NET 7 on May 2024

**Comment by hishamco** [comment]
I'm not sure why we need to drop the support for both .NET 6 & 7? OC supports multi-targeting from the beginning

We need to drop .NET 7 on May 2024

**Comment by hishamco** [comment]
I'm not sure why we need to drop the support for both .NET 6 & 7? OC supports multi-targeting from the beginning

We need to drop .NET 7 on May 2024

**Comment by Piedone** [comment]
Our practice has always been to support the latest LTS version of .NET, and additionally, the latest version if it's different. But please add your arguments under https://github.com/OrchardCMS/OrchardCore/issues/14694.

**Comment by sebastienros** [comment]
> build warnings need to be fixed in YesSql: https://github.com/sebastienros/yessql/issues/510.

You can also reference a specific version explicitly. Though it's possible that it's binary incompatible since it's a major version bump, it might also just work.

**Comment by Piedone** [comment]
Do you mean referencing `SQLitePCLRaw.lib.e_sqlite3` directly and thus trying to load overriding YesSql's reference?

**Comment by MikeAlhayek** [comment]
> I'm not sure why we need to drop the support for both .NET 6 & 7? OC supports multi-targeting from the beginning

Because we want to use API's that are not available in .NET 6 and possible .NET 7.

**Review by MikeAlhayek — APPROVED** [review]

**Comment by hishamco** [comment]
> Because we want to use API's that are not available in .NET 6 and possible .NET 7.

You can do that as we did it before with preprocessor directive

**Comment by sebastienros** [comment]
@Piedone just add a direct reference to Microsoft.Data.Sqlite version 8.0 in the project that references YesSql. Then run the tests on Sqlite. But I will approve the PR, just try if you are curious.

File: src/OrchardCore.Build/Dependencies.AspNetCore.props
Comment (sebastienros) [inline src/OrchardCore.Build/Dependencies.AspNetCore.props:27]: WE need to keep it commented as an example, otherwise we'll forget how to do that when net9.0 ships. We had this problem last time and the templates got broken.

File: .github/workflows/pr_ci.yml
Comment (sebastienros) [inline .github/workflows/pr_ci.yml:29]: I am wondering if not setting `-f` would have just built all tfms.

**Review by sebastienros — APPROVED** [review]

**Comment by MikeAlhayek** [comment]
> Do you mean referencing SQLitePCLRaw.lib.e_sqlite3 directly and thus trying to load overriding YesSql's reference?

PR #14706 will be merged soon, this will will resolve the issue. Once it's merged, you may rebase your branch and the warnings should go away

File: .github/workflows/pr_ci.yml
Comment (jtkech) [reply .github/workflows/pr_ci.yml:29]: Yes but when multiple tfms are built concurrently there are some issues when it outputs files.

File: src/OrchardCore.Build/Dependencies.AspNetCore.props
Comment (Piedone) [reply src/OrchardCore.Build/Dependencies.AspNetCore.props:27]: Re-added it with a comment.

**Comment by Piedone** [comment]
The build warnings are indeed gone, thank you!

**Comment by ns8482e** [comment]
Not sure if this is the the right move for OC to remove support of .NET 6 (November 12, 2024) and .NET 7 (May 14, 2024).

**Comment by MikeAlhayek** [comment]
@ns8482e is there a real reason to stay on .net6?

**Comment by hishamco** [comment]
I'm with @ns8482e we might not care about .NET6.0 but we can about .NET 7.0 support, the current plan forces everyone to upgrade to .NET 8.0

**Comment by ns8482e** [comment]
@MikeAlhayek  IMO keeping .NET 6 support allow OC users to upgrade to .NET 8 at-will  and will still get the bug fixes. 
We can always use new API's for new features, using `NET8_0_OR_GREATER`, `NET7_0_OR_GREATER` while still supprting `.NET 6`

**Comment by ns8482e** [comment]
@hishamco  should care about .NET 6 as it has longer support life than .NET 7 and mostly slow moving corporates only upgrade to LTS versions

**Comment by hishamco** [comment]
1 year is not bad I though it will be end within two months ;)

**Comment by MikeAlhayek** [comment]
@ns8482e is there a reason people should not upgrade .net 6 to 8? 

https://github.com/OrchardCMS/OrchardCore/issues/14694#issuecomment-1815019602

**Comment by ns8482e** [comment]
> @ns8482e is there a reason people should not upgrade .net 6 to 8?
> 
> [#14694 (comment)](https://github.com/OrchardCMS/OrchardCore/issues/14694#issuecomment-1815019602)

No real reason :) -  people are just skeptical and lazy to switch to new tech and prefer to stay on LTS until support ended.

**Comment by MikeAlhayek** [comment]
Giving lazy people more choice just restricts us 😀.

**Comment by sebastienros** [comment]
Supporting net6.0 prevents us from removing NewtonSoft.Json. It should be as easy as updating the target framework in the websites that decide to upgrade to the next OC version. Please try on your site with the main branch and if there is any build or runtime issue we can't fix then we can add it back.

**Pull request merged**
